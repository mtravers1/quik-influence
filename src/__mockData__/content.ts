const data = {
  homepage: {
    banner: {
      resource: 'quik-influence',
      page: 'Home Page',
      type: 'banner',
      content: {
        sub_header: 'QUIK INFLUENCE THE CAPABLE CRM',
        header: 'Find your customer with Quik Influence',
        header_desc: 'new desc',
        image_url: 'url',
      },
    },
  },
};

export default data;
