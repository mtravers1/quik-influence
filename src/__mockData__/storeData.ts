const initialState: any = {
  auth: {
    user: { admin: { avatar: '', firstName: '', lastName: '' } },
    loading: false,
  },
  nav: null,
  campaigns: {
    campaigns: null,
    loading: true,
  },
};

export default initialState;
