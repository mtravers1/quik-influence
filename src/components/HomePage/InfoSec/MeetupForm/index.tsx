import { Box, Flex } from '@chakra-ui/react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { FormControl, FormErrorMessage } from '@chakra-ui/react';
import Input from '../../Input';
import CustomButton from 'components/Button';
import useInput from 'hooks/useInput';
import formdata from 'utils/constants/formData/meetTeam';

const MeetupForm = () => {
  const { handleChange, inputTypes, handleSubmit, errors } = useInput({
    inputs: formdata.map(data => ({
      name: data.name,
      required: data.required,
    })),
    cb: async inputs => {
      // do what you will with inputs
      console.log('Submitted');
    },
  });

  return (
    <Flex
      display="flex"
      as="form"
      onClick={handleSubmit}
      flexDirection="column"
      flexGrow={1}
      alignItems="center"
    >
      <Flex flexWrap="wrap" marginBottom={30} justifyContent="space-between">
        {formdata.map((data, i) => (
          <FormControl
            key={`contact_form_${i}`}
            w="48%"
            isInvalid={errors[data.name]}
            isRequired={data.required}
            margin="3px 0"
          >
            <Box position={'relative'}>
              {' '}
              <Input
                name={data.name}
                placeholder={data.label}
                onChange={handleChange}
                value={inputTypes[data.name]}
                datatest-id={`test-${data.name}`}
              />
              <FontAwesomeIcon
                style={{
                  width: '10px',
                  position: 'absolute',
                  top: '50%',
                  left: '30px',
                  transform: 'translateY(-50%)',
                  zIndex: 1,
                }}
                icon={data.icon}
                color="red"
              />
            </Box>

            {errors[data.name] && (
              <FormErrorMessage paddingLeft={50} fontSize={12}>
                {data.errorMessage}
              </FormErrorMessage>
            )}
          </FormControl>
        ))}
      </Flex>

      <CustomButton
        borderRadius={40}
        maxWidth={320}
        fontSize={14}
        paddingTop={23}
        paddingBottom={23}
      >
        CONTACT US
      </CustomButton>
    </Flex>
  );
};

export default MeetupForm;
