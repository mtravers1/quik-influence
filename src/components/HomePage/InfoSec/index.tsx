import EditableWrapper from 'components/EditableWrapper';
import Section from '../Section';
import { Flex, Image } from '@chakra-ui/react';

const InfoSec = ({ info }: { info: any }) => {
  return (
    <Section
      background="#fff"
      h="unset"
      padding={{ base: '50px 15px', lg: '70px 15px', xl: '100px 15px' }}
    >
      <Flex
        maxW="1440px"
        margin="auto"
        alignItems="center"
        justifyContent="center"
        h="100%"
        direction="column"
      >
        <EditableWrapper
          sectionId="header"
          data={info}
          sectionName="info"
          as="h2"
          color="#333"
          maxW={{ base: '90%', lg: '600px', xl: '736px' }}
          textAlign="center"
          margin="0 auto 10px"
          fontSize={{ base: '25px', lg: '30px', xl: '43px' }}
        >
          {info.content.header}
        </EditableWrapper>

        <EditableWrapper
          sectionId="sub_header"
          data={info}
          sectionName="info"
          as="p"
          color="#333"
          textAlign="center"
          margin="0 auto 35px"
          maxW={{ base: '70%', lg: '500px', xl: '611px' }}
        >
          {info.content.sub_header}
        </EditableWrapper>

        <Flex
          justifyContent="space-between"
          marginBottom="32px"
          maxW={1000}
          textAlign="center"
          direction={{ base: 'column', md: 'row' }}
          alignItems="center"
        >
          {info.content.cards.map(
            (
              card: { header: string; desc: string; img: string },
              i: number
            ) => (
              <Flex
                key={`info_box_${i}`}
                border="1px solid #E6E6E6"
                alignItems="center"
                justifyContent="center"
                direction="column"
                p="30px 15px"
                w={{ base: '100%', xl: '32%' }}
                margin={{ base: '20px', xl: 0 }}
                maxW="450px"
                bg={i % 2 === 1 ? 'red' : '#fff'}
              >
                <EditableWrapper
                  sectionId={`cards__${i}__header`}
                  data={info}
                  sectionName="info"
                  as="h3"
                  fontSize={25}
                  color={i % 2 === 1 ? '#fff' : '#333'}
                  fontWeight="500"
                >
                  {card.header}
                </EditableWrapper>

                <EditableWrapper
                  sectionId={`cards__${i}__img`}
                  data={info}
                  sectionName="info"
                  isImage
                >
                  <Image margin="25px 0" src={card.img} w="75px" h="75px" />
                </EditableWrapper>

                <EditableWrapper
                  sectionId={`cards__${i}__desc`}
                  data={info}
                  sectionName="info"
                  as="p"
                  color={i % 2 === 1 ? '#fff' : '#333'}
                >
                  {card.desc}
                </EditableWrapper>
              </Flex>
            )
          )}
        </Flex>

        <EditableWrapper
          sectionId="footer"
          data={info}
          sectionName="info"
          as="p"
          fontSize={14}
          fontWeight="500"
          color="#333"
        >
          {info.content.footer}
        </EditableWrapper>
      </Flex>
    </Section>
  );
};

export default InfoSec;
