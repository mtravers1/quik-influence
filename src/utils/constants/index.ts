export const CONTENT_URL = '/content';
export const Q_TOKEN = '_q_inf';
export const DEFAULT_PAGE_SIZE = '100';
export const ADMINS_ID = [2];

export const FILE_STATUS = {
  PENDING: 'pending',
  PROCESSING: 'processing',
  UPLOADING: 'uploading',
  PAUSED: 'paused',
  COMPLETED: 'completed',
  FAILED: 'failed',
  IDLE: 'idle',
};

export const headers: string[] = [
  'phone',
  'avatar',
  'email',
  'firstName',
  'lastName',
  'address',
  'city',
  'dateOfBirth',
  'instagramId',
  'socialMediaHandle',
  'state',
  'country',
  'postalCode',
  'gender',
  'address2',
  'address3',
  'facebookHandle',
  'tiktokHandle',
  'twitterHandle',
];

export const formDataRelatedToSpecificUser = [
  'updatedAt',
  'twitterHandle',
  'avatar',
  'phone',
  'address2',
  'email',
  'address1',
  'postalCode',
  'socialMediaHandle',
  'lastName',
  'firstName',
  'instagramId',
  'tiktokHandle',
  'address',
  'address3',
  'facebookHandle',
];

export const MARKETING_ADMIN = ['can_approve_or_reject_campaign'];
export const ADMIN_USER_AFFILIATE = ['can_get_joinable_campaigns'];
export const CAN_HANDLE_INFLUENCERS = ['can_manage_influencers'];

export const MARKET_PLACE_NEW_PRODUCTS = 'new';
export const MARKET_PLACE_POPULAR_PRODUCTS = 'popular';

export const CART_CLICK_NAME = 'cart';
