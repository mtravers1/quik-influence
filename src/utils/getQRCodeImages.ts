import QRCode, { QRCodeToDataURLOptions } from 'qrcode';
import { QrCodeRawData } from 'types';

const defaultOptions: Partial<QRCodeToDataURLOptions> = {
  errorCorrectionLevel: 'M',
  width: 256,
};

export default async function getQRImages(
  list: QrCodeRawData[],
  options?: QRCodeToDataURLOptions,
): Promise<string[]> {
  const qrCodeUrlPromises = list.map(({ text }) =>
    QRCode.toDataURL(text, { ...defaultOptions, ...options }),
  );
  const images = await Promise.all(qrCodeUrlPromises);
  return images;
}
