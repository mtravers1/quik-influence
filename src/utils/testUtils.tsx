import { FC, ReactElement } from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from 'redux/reducers';

function makeTestStore(rootReducer: any, initialState: any) {
  const store = createStore(rootReducer, initialState, applyMiddleware(thunk));
  const origDispatch = store.dispatch;
  store.dispatch = jest.fn(origDispatch);
  return store;
}

const renderWithStore = (ui: ReactElement, storedefaults: any) => {
  const { initialState, ...renderOptions } = storedefaults;
  const store = makeTestStore(rootReducer, initialState);

  const Wrapper: FC = ({ children }) => {
    return <Provider store={store}>{children}</Provider>;
  };

  return render(ui, { wrapper: Wrapper, ...renderOptions });
};

export * from '@testing-library/react';
export { renderWithStore };
