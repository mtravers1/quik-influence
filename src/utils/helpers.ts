import axios from 'axios';
import { omitBy, isNil } from 'lodash';
import { ADMINS_ID, Q_TOKEN } from './constants';

import { DropdownSelectOption } from 'components/DropdownSelect';
import { FilterDataProps } from 'types';
import { format } from 'date-fns';

export const baseurl = process.env.BACKEND_URL;

export const axiosInstance = axios.create({
  baseURL: `${baseurl}/api/v1`,
  // withCredentials: true,
  headers: {
    'Access-Control-Allow-Headers':
      'Access-Control-Allow-Origin, Access-Control-Allow-Headers, Content-Type',
    'Access-Control-Allow-Origin': '*',
  },
});

export const logout = () => {
  localStorage.removeItem(Q_TOKEN);
  window.location.href = '/login';
};

export const validate = (field: any, pattern: any) => {
  const parts = /\/(.*)\/(.*)/.exec(pattern) || [];
  let restoredRegex = new RegExp(parts[1], parts[2]);
  if (restoredRegex.test(field)) return true;
  return false;
};

export const setToken = (token: string) => {
  axiosInstance.defaults.headers.common['token'] = token;

  if (typeof window !== 'undefined') {
    localStorage.setItem(Q_TOKEN, token);
  }
};

const tokens: any = {};

export function parseJwt(token: any) {
  if (!token) return;
  if (tokens[token]) return tokens[token];

  const base64Url = token.split('.')[1];
  const base64 = base64Url.replace(/-/g, '+').replace(/_/g, '/');
  const jsonPayload = decodeURIComponent(
    atob(base64)
      // Buffer.from(base64, 'base64')
      .split('')
      .map(c => {
        return '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2);
      })
      .join('')
  );

  const result = JSON.parse(jsonPayload);
  tokens[token] = result;

  return result;
}

export function getUser() {
  let user;
  let ctoken;
  let isExpired: boolean = false;

  if (typeof window !== 'undefined') {
    ctoken = localStorage.getItem(Q_TOKEN);
  }

  if (ctoken !== 'null') {
    user = parseJwt(ctoken);
  }

  isExpired = user && user.exp && user.exp < Date.now() / 1000;

  return { admin: user, token: ctoken, isExpired };
}

export const getNumberRange = (
  start: number,
  stop: number,
  step: number
): DropdownSelectOption[] =>
  Array.from({ length: (stop - start) / step + 1 }, (_, i) => ({
    label: (start + i * step).toString(),
    value: (start + i * step).toString(),
  }));

export const getQueryString = (params?: FilterDataProps) => {
  const paramsFilters = omitBy(params, isNil);
  const query = Object.keys(paramsFilters)
    .map(
      k => encodeURIComponent(k) + '=' + encodeURIComponent(paramsFilters[k])
    )
    .join('&');
  return query;
};

export const getSocialHandleHeader = (socialColumns: string[]): string[] => {
  let socialHeader: string[] = [];
  socialColumns.forEach((socialColumn: any) => {
    switch (socialColumn) {
      case 'facebookHandle':
        socialHeader.push('Facebook');
        break;
      case 'twitterHandle':
        socialHeader.push('Twitter');
        break;
      case 'instagramId':
        socialHeader.push('Instagram');
        break;
      case 'tiktokHandle':
        socialHeader.push('Tik Tok');
        break;
      default:
        break;
    }
  });
  return socialHeader;
};

export const similarity = (s1: string, s2: string) => {
  let longer = s1;
  let shorter = s2;
  if (s1.length < s2.length) {
    longer = s2;
    shorter = s1;
  }
  let longerLength = longer.length;
  if (longerLength == 0) {
    return 1.0;
  }
  return (longerLength - editDistance(longer, shorter)) / longerLength;
};

function editDistance(s1: string, s2: string) {
  s1 = s1.toLowerCase();
  s2 = s2.toLowerCase();

  let costs = new Array();
  for (let i = 0; i <= s1.length; i++) {
    let lastValue = i;
    for (let j = 0; j <= s2.length; j++) {
      if (i == 0) costs[j] = j;
      else {
        if (j > 0) {
          let newValue = costs[j - 1];
          if (s1.charAt(i - 1) != s2.charAt(j - 1))
            newValue = Math.min(Math.min(newValue, lastValue), costs[j]) + 1;
          costs[j - 1] = lastValue;
          lastValue = newValue;
        }
      }
    }
    if (i > 0) costs[s2.length] = lastValue;
  }
  return costs[s2.length];
}

export const stringSearch = (val: string, string: string) =>
  string && string.toLowerCase().search(val.toLowerCase()) !== -1;

export function isInViewport(element: any) {
  const rect = element.getBoundingClientRect();
  return (
    rect.top >= 0 &&
    rect.left >= 0 &&
    rect.bottom <=
      (window.innerHeight || document.documentElement.clientHeight) &&
    rect.right <= (window.innerWidth || document.documentElement.clientWidth)
  );
}

export const isAdmin = () => {
  const user = getUser();
  return user.admin && ADMINS_ID.includes(user.admin.roleId);
};

export const hasPermission = (
  userPerm: string[] | undefined,
  permission: string[] | undefined
) => {
  if (userPerm?.length === 0) return true;
  return userPerm?.some(perm => permission?.includes(perm));
};

export const getReportOpenDate = (events: any) => {
  if (!events) return 'NA';
  const openEvent = events.find((evt: any) => evt['event_name'] === 'open');
  if (!openEvent) return 'Not opened yet';

  return format(new Date(openEvent['processed']), 'dd/MM/yyyy');
};

export const getReportUnsub = (events: any) => {
  if (!events) return 'NA';
  const unsubEvent = events.find(
    (evt: any) => evt['event_name'] === 'unsubscribe'
  );
  if (!unsubEvent) return 'No';

  return 'Yes';
};

export const truncateText = (str: string, num: number) => {
  if (str.length <= num) {
    return str;
  }
  return str.slice(0, num) + '...';
};
