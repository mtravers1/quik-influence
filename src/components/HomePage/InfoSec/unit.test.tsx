import { renderWithStore } from 'utils/testUtils';
import { staticContentData } from '__mockData__/content';
import initialState from '__mockData__/storeData';
import InfoSec from '.';

it('renders MainContent component', () => {
  const { container } = renderWithStore(
    <InfoSec info={staticContentData.props.pageContent.info} />,
    { initialState }
  );
  expect(container).toMatchSnapshot();
});
