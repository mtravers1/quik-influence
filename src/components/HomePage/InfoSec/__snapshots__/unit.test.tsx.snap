// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainContent component 1`] = `
<div>
  <section
    class="css-15a0p7b"
  >
    <div
      class="css-p7r2ak"
    >
      <div
        class="css-79elbk"
      >
        <div
          class="css-g90ssd"
        >
          <img
            alt="Close"
            src="/close.png"
            style="width: 10px; height: 10px;"
          />
        </div>
        <h2
          class="css-1tpxf1l"
          contenteditable="false"
          id="info_header"
          tabindex="-1"
        >
          Our Specialized apps and services make customer acquisition easy
        </h2>
      </div>
      <div
        class="css-79elbk"
      >
        <div
          class="css-g90ssd"
        >
          <img
            alt="Close"
            src="/close.png"
            style="width: 10px; height: 10px;"
          />
        </div>
        <p
          class="css-1h356hx"
          contenteditable="false"
          id="info_sub_header"
          tabindex="-1"
        >
          We craft unique digital experiences, Because of this we have an endless supply of leads all organic and verified.
        </p>
      </div>
      <div
        class="css-b69u32"
      >
        <div
          class="css-rxw4r2"
        >
          <div
            class="css-79elbk"
          >
            <div
              class="css-g90ssd"
            >
              <img
                alt="Close"
                src="/close.png"
                style="width: 10px; height: 10px;"
              />
            </div>
            <h3
              class="css-1mws978"
              contenteditable="false"
              id="info_cards__0__header"
              tabindex="-1"
            >
              Industry Experts
            </h3>
          </div>
          <img
            class="chakra-image css-20teaf"
            src="/icon.png"
          />
          <div
            class="css-79elbk"
          >
            <div
              class="css-g90ssd"
            >
              <img
                alt="Close"
                src="/close.png"
                style="width: 10px; height: 10px;"
              />
            </div>
            <p
              class="css-1yxu09j"
              contenteditable="false"
              id="info_cards__0__desc"
              tabindex="-1"
            >
              Not sure what your looking for? Our team can help you dial in on what will be your next audience.
            </p>
          </div>
        </div>
        <div
          class="css-qvkken"
        >
          <div
            class="css-79elbk"
          >
            <div
              class="css-g90ssd"
            >
              <img
                alt="Close"
                src="/close.png"
                style="width: 10px; height: 10px;"
              />
            </div>
            <h3
              class="css-14oezjm"
              contenteditable="false"
              id="info_cards__1__header"
              tabindex="-1"
            >
              100's of industries
            </h3>
          </div>
          <img
            class="chakra-image css-20teaf"
            src="/icon.png"
          />
          <div
            class="css-79elbk"
          >
            <div
              class="css-g90ssd"
            >
              <img
                alt="Close"
                src="/close.png"
                style="width: 10px; height: 10px;"
              />
            </div>
            <p
              class="css-16qv2i2"
              contenteditable="false"
              id="info_cards__1__desc"
              tabindex="-1"
            >
              Our leads are sourced from a Multi-Channel system that spans across many different platforms and industries
            </p>
          </div>
        </div>
        <div
          class="css-rxw4r2"
        >
          <div
            class="css-79elbk"
          >
            <div
              class="css-g90ssd"
            >
              <img
                alt="Close"
                src="/close.png"
                style="width: 10px; height: 10px;"
              />
            </div>
            <h3
              class="css-1mws978"
              contenteditable="false"
              id="info_cards__2__header"
              tabindex="-1"
            >
              Custom experience
            </h3>
          </div>
          <img
            class="chakra-image css-20teaf"
            src="/icon.png"
          />
          <div
            class="css-79elbk"
          >
            <div
              class="css-g90ssd"
            >
              <img
                alt="Close"
                src="/close.png"
                style="width: 10px; height: 10px;"
              />
            </div>
            <p
              class="css-1yxu09j"
              contenteditable="false"
              id="info_cards__2__desc"
              tabindex="-1"
            >
              Our amazing development team can create an expereince tailored to your comapnies needs. No matter what.
            </p>
          </div>
        </div>
      </div>
      <div
        class="css-79elbk"
      >
        <div
          class="css-g90ssd"
        >
          <img
            alt="Close"
            src="/close.png"
            style="width: 10px; height: 10px;"
          />
        </div>
        <p
          class="css-crqql1"
          contenteditable="false"
          id="info_footer"
          tabindex="-1"
        >
          Contact us about custom plans and agreements to ensure we meet your specific needs!
        </p>
      </div>
    </div>
  </section>
</div>
`;
